unable to build Windows app

https://brendanforster.visualstudio.com/desktop-linux/_build/results?buildId=422

```
$ ts-node -P script/tsconfig.json script/post-install.ts
yarn install v1.17.3
[1/4] Resolving packages...
[2/4] Fetching packages...
[3/4] Linking dependencies...
[4/4] Rebuilding all packages...
error D:\a\1\s\app\node_modules\keytar: Command failed.
Exit code: 1
Command: prebuild-install || node-gyp rebuild
Arguments: 
Directory: D:\a\1\s\app\node_modules\keytar
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
Output:
D:\a\1\s\app\node_modules\node-abi\index.js:30
  throw new Error('Could not detect abi for version ' + target + ' and runtime ' + runtime + '.  Updating "node-abi" might help solve this issue if it is a new release of ' + runtime)
  ^

Error: Could not detect abi for version 5.0.6 and runtime electron.  Updating "node-abi" might help solve this issue if it is a new release of electron
    at getAbi (D:\a\1\s\app\node_modules\node-abi\index.js:30:9)
    at module.exports (D:\a\1\s\app\node_modules\prebuild-install\rc.js:71:57)
    at Object.<anonymous> (D:\a\1\s\app\node_modules\prebuild-install\bin.js:9:25)
    at Module._compile (internal/modules/cjs/loader.js:955:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:991:10)
    at Module.load (internal/modules/cjs/loader.js:811:32)
    at Function.Module._load (internal/modules/cjs/loader.js:723:14)
    at Function.Module.runMain (internal/modules/cjs/loader.js:1043:10)
    at internal/main/run_main_module.js:17:11

D:\a\1\s\app\node_modules\keytar>if not defined npm_config_node_gyp (node "C:\hostedtoolcache\windows\node\12.14.1\x64\node_modules\npm\bin\node-gyp-bin\\..\..\node_modules\node-gyp\bin\node-gyp.js" rebuild )  else (node "" rebuild ) 
gyp info it worked if it ends with ok
gyp info using node-gyp@5.0.5
gyp info using node@12.14.1 | win32 | x64
gyp info find Python using Python version 2.7.9 found at "C:/ProgramData/Chocolatey/lib/mingw/tools/install/mingw64/opt/bin/python2.7.exe"
gyp http GET https://atom.io/download/electron/v5.0.6/node-v5.0.6-headers.tar.gz
gyp http 200 https://atom.io/download/electron/v5.0.6/node-v5.0.6-headers.tar.gz
gyp http GET https://atom.io/download/electron/v5.0.6/SHASUMS256.txt
gyp http GET https://atom.io/download/electron/v5.0.6/win-x86/node.lib
gyp http GET https://atom.io/download/electron/v5.0.6/win-x64/node.lib
gyp http GET https://atom.io/download/electron/v5.0.6/win-arm64/node.lib
gyp http 200 https://atom.io/download/electron/v5.0.6/SHASUMS256.txt
gyp http 403 https://atom.io/download/electron/v5.0.6/win-arm64/node.lib
gyp WARN install got an error, rolling back install
gyp http 200 https://atom.io/download/electron/v5.0.6/win-x86/node.lib
gyp http 200 https://atom.io/download/electron/v5.0.6/win-x64/node.lib
gyp ERR! configure error 
gyp ERR! stack Error: 403 status code downloading arm64 node.lib
gyp ERR! stack     at Request.<anonymous> (C:\hostedtoolcache\windows\node\12.14.1\x64\node_modules\npm\node_modules\node-gyp\lib\install.js:335:22)
gyp ERR! stack     at Request.emit (events.js:228:7)
gyp ERR! stack     at Request.onRequestResponse (C:\hostedtoolcache\windows\node\12.14.1\x64\node_modules\npm\node_modules\request\request.js:1066:10)
gyp ERR! stack     at ClientRequest.emit (events.js:223:5)
gyp ERR! stack     at HTTPParser.parserOnIncomingClient [as onIncoming] (_http_client.js:583:27)
gyp ERR! stack     at HTTPParser.parserOnHeadersComplete (_http_common.js:115:17)
gyp ERR! stack     at TLSSocket.socketOnData (_http_client.js:456:22)
gyp ERR! stack     at TLSSocket.emit (events.js:223:5)
gyp ERR! stack     at addChunk (_stream_readable.js:309:12)
gyp ERR! stack     at readableAddChunk (_stream_readable.js:290:11)
gyp ERR! System Windows_NT 10.0.14393
gyp ERR! command "C:\\hostedtoolcache\\windows\\node\\12.14.1\\x64\\node.exe" "C:\\hostedtoolcache\\windows\\node\\12.14.1\\x64\\node_modules\\npm\\node_modules\\node-gyp\\bin\\node-gyp.js" "rebuild"
gyp ERR! cwd D:\a\1\s\app\node_modules\keytar
gyp ERR! node -v v12.14.1
gyp ERR! node-gyp -v v5.0.5
gyp ERR! not ok
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
##[error]Cmd.exe exited with code '1'.
##[section]Finishing: Install
```

-

--- a/app/test/unit/git/worktree-test.ts
+++ b/app/test/unit/git/worktree-test.ts
@@ -10,7 +10,7 @@ import {
   cleanupTemporaryWorkTrees,
 } from '../../../src/lib/git/worktree'
 import { Repository, LinkedWorkTree } from '../../../src/models/repository'
-import { realpathSync } from 'fs-extra'
+import { realpathSync, mkdtemp } from 'fs-extra'
 
 describe('git/worktree', () => {
   describe('listWorktrees', () => {
@@ -18,7 +18,30 @@ describe('git/worktree', () => {
       let repository: Repository
 
       beforeEach(async () => {
-        repository = await setupEmptyRepository()
+        if (
+          process.platform === 'win32' &&
+          process.env.TF_BUILD != null &&
+          process.env.AGENT_TEMPDIRECTORY
+        ) {
+          // running test on Azure Pipelines (Windows)
+          //
+          // This is a workaround for the `TEMP` environment variable containing
+          // a shortname (VSSADM~1) version of the account name, but the NodeJS
+          // API returning the long path name (VSSAdministrator). Both are valid
+          // but will break when we try to compare them.
+          //
+          // Instead of using `TEMP` in this situation we'll use the
+          // `AGENT_TEMPDIRECTORY` environment variable which will be cleaned up
+          // after the test run
+          const repoPath = await mkdtemp(
+            Path.join(process.env.AGENT_TEMPDIRECTORY, 'desktop-empty-repo-')
+          )
+          await GitProcess.exec(['init'], repoPath)
+
+          repository = new Repository(repoPath, -1, null, false)
+        } else {
+          repository = await setupEmptyRepository()
+        }
       })
 
       it('returns one entry', async () => {

--- a/azure-pipelines.yml
+++ b/azure-pipelines.yml
@@ -31,7 +31,7 @@ jobs:
     steps:
       - task: NodeTool@0
         inputs:
-          versionSpec: '12.x'
+          versionSpec: '10.17.x'
       - task: geeklearningio.gl-vsts-tasks-yarn.yarn-installer-task.YarnInstaller@2
         inputs:
           versionSpec: '1.5.1'
@@ -121,7 +121,7 @@ jobs:
     steps:
       - task: NodeTool@0
         inputs:
-          versionSpec: '12.x'
+          versionSpec: '10.17.x'
       - script: |
           yarn install --force
         name: Install
@@ -153,7 +153,7 @@ jobs:
           sudo apt-get install -y --no-install-recommends libsecret-1-dev
       - task: NodeTool@0
         inputs:
-          versionSpec: '12.x'
+          versionSpec: '10.17.x'
       - task: geeklearningio.gl-vsts-tasks-yarn.yarn-installer-task.YarnInstaller@2
         inputs:
           versionSpec: '1.5.1'

